flink-connector-prometheus/src/main/java/org/apache/flink/connector/prometheus/table/RowDataToPrometheusTimeSeriesConverter.java

```diff
@@ -0,0 +1,116 @@
+public class RowDataToPrometheusTimeSeriesConverter {
+
+    private final DataType physicalDataType;
+    private final PrometheusConfig prometheusConfig;
+
+    public RowDataToPrometheusTimeSeriesConverter(
+            DataType physicalDataType, PrometheusConfig prometheusConfig) {
+        this.physicalDataType = physicalDataType;
+        this.prometheusConfig = prometheusConfig;
+    }
+
+    public PrometheusTimeSeries convertRowData(RowData row) {
+        List<DataTypes.Field> fields = DataType.getFields(physicalDataType);
+
+        PrometheusTimeSeries.Builder builder = PrometheusTimeSeries.builder();
+        Double sampleValue = null;
+        Long sampleTimestamp = null;
+
+        for (int i = 0; i < fields.size(); i++) {
+            DataTypes.Field field = fields.get(i);
+            RowData.FieldGetter fieldGetter =
+                    createFieldGetter(fields.get(i).getDataType().getLogicalType(), i);
+            FieldValue fieldValue = new FieldValue(fieldGetter.getFieldOrNull(row));
+            String fieldName = field.getName();
+
+            if (fieldName.equals(prometheusConfig.getMetricName())) {
+                builder.withMetricName(fieldValue.getStringValue());
+            } else if (fieldName.equals(prometheusConfig.getMetricSampleKey())) {
+                sampleValue = fieldValue.getDoubleValue();
+            } else if (prometheusConfig.getLabelKeys().contains(fieldName)) {
+                builder.addLabel(fieldName, fieldValue.getStringValue());
+            } else if (fieldName.equals(prometheusConfig.getMetricSampleTimestamp())) {
+                sampleTimestamp = fieldValue.getLongValue();
+            } else {
+                throw new IllegalArgumentException(
+                        String.format(
+                                "%s is not a supported field, valid fields are: %s",
+                                fieldName,
+                                Arrays.asList(
+                                        prometheusConfig.getMetricName(),
+                                        prometheusConfig.getLabelKeys(),
+                                        prometheusConfig.getMetricSampleKey(),
+                                        prometheusConfig.getMetricSampleTimestamp())));
+            }
+        }
+
+        if (sampleValue != null && sampleTimestamp != null) {
+            builder.addSample(sampleValue, sampleTimestamp);
+        } else {
+            throw new IllegalArgumentException(
+                    String.format(
+                            "Row is missing sampleValue field; %s or sampleTimestamp: %s",
+                            sampleValue, sampleTimestamp));
+        }
+
+        return builder.build();
+    }
+
+    private static class FieldValue {
+        private final Object value;
+
+        private FieldValue(Object value) {
+            this.value = value;
+        }
+
+        private String getStringValue() {
+            return value.toString();
+        }
```

Comment:
Are there any unsupported types? What happens if the field provided is a `MAP` / `ARRAY` ? 
https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/types/#list-of-data-types

Reply:
This is a good call out, we only support `org.apache.flink.table.data.StringData` type here, have added validation for this in latest revision